source/audio.py

```diff
@@ -0,0 +1,210 @@
+		return {
+			SoundSplitState.OFF: (1.0, 1.0),
+			SoundSplitState.NVDA_LEFT: (0.0, 1.0),
+			SoundSplitState.NVDA_RIGHT: (1.0, 0.0),
+		}[self]
+	
+	def getNVDAVolume(self) -> VolumeTupleT:
+		return {
+			SoundSplitState.OFF: (1.0, 1.0),
+			SoundSplitState.NVDA_LEFT: (1.0, 0.0),
+			SoundSplitState.NVDA_RIGHT: (0.0, 1.0),
+		}[self]
+
+
+@unique
+class SoundSplitToggleMode(DisplayStringIntEnum):
+	OFF_LEFT_RIGHT = 0
+	OFF_AND_NVDA_LEFT = 1
+	OFF_AND_NVDA_RIGHT = 2
+
+	@property
+	def _displayStringLabels(self) -> Dict[IntEnum, str]:
+		return {
+			# Translators: Sound split toggle mode
+			SoundSplitToggleMode.OFF_AND_NVDA_LEFT: _("Cycles through off and NVDA on the left"),
+			# Translators: Sound split toggle mode
+			SoundSplitToggleMode.OFF_AND_NVDA_RIGHT: _("Cycles through off and NVDA on the right"),
+			# Translators: Sound split toggle mode
+			SoundSplitToggleMode.OFF_LEFT_RIGHT: _("Cycles through off, NVDA on the left and NVDA on the right"),
+		}
+	
+	def getPossibleStates(self) -> List[SoundSplitState]:
+		result = [SoundSplitState.OFF]
+		if 'LEFT' in self.name:
+			result.append(SoundSplitState.NVDA_LEFT)
+		if 'RIGHT' in self.name:
+			result.append(SoundSplitState.NVDA_RIGHT)
+		return result
+	
+	def getClosestState(self, state: SoundSplitState) -> SoundSplitState:
+		states = self.getPossibleStates()
+		if state in states:
+			return state
+		return states[-1]
+
+
+sessionManager: audiopolicy.IAudioSessionManager2 = None
+activeCallback: Optional[comtypes.COMObject] = None
+
+
+def initialize() -> None:
+	global sessionManager
+	sessionManager = getSessionManager()
+	if sessionManager is None:
+		log.error("Could not initialize audio session manager! ")
+		return
+	if nvwave.usingWasapiWavePlayer():
+		state = SoundSplitState(config.conf['audio']['soundSplitState'])
+		global activeCallback
+		activeCallback = setSoundSplitState(state)
+
+
+@atexit.register
+def terminate():
+	if nvwave.usingWasapiWavePlayer():
+		setSoundSplitState(SoundSplitState.OFF)
+		if activeCallback is not None:
+			unregisterCallback(activeCallback)
+
+
+def getDefaultAudioDevice(kind: EDataFlow = EDataFlow.eRender) -> Optional[mmdeviceapi.IMMDevice]:
+	deviceEnumerator = comtypes.CoCreateInstance(
+		CLSID_MMDeviceEnumerator,
+		mmdeviceapi.IMMDeviceEnumerator,
+		comtypes.CLSCTX_INPROC_SERVER,
+	)
+	device = deviceEnumerator.GetDefaultAudioEndpoint(
+		kind.value,
+		ERole.eMultimedia.value,
+	)
+	return device
+
+
+def getSessionManager() -> audiopolicy.IAudioSessionManager2:
+	audioDevice = getDefaultAudioDevice()
+	if audioDevice is None:
+		raise RuntimeError("No default output audio device found!")
+	tmp = audioDevice.Activate(audiopolicy.IAudioSessionManager2._iid_, comtypes.CLSCTX_ALL, None)
+	sessionManager: audiopolicy.IAudioSessionManager2 = tmp.QueryInterface(audiopolicy.IAudioSessionManager2)
+	return sessionManager
+
+
+def applyToAllAudioSessions(
+		func: Callable[[audiopolicy.IAudioSessionControl2], NoReturn],

```

Comment:
`NoReturn` is not a right type here. As per the [documentation](https://docs.python.org/3.11/library/typing.html#typing.NoReturn) it should be used for function which either should never be called , or for those which unconditionally raise an exception. This should just have `None` as a return annotation.